pallets/capacity/src/lib.rs

```diff
@@ -477,6 +715,36 @@ impl<T: Config> Pallet<T> {
 		Ok(capacity)
 	}
 
+	fn increase_stake_and_issue_boost_capacity(
+		staker: &T::AccountId,
+		staking_details: &mut StakingDetails<T>,
+		target: &MessageSourceId,
+		amount: &BalanceOf<T>,
+	) -> Result<BalanceOf<T>, DispatchError> {
+		staking_details.deposit(*amount).ok_or(ArithmeticError::Overflow)?;
+		Self::set_staking_account_and_lock(staker, staking_details)?;
+
+		// get the capacity generated by a Provider Boost
+		let capacity = Self::capacity_generated(T::RewardsProvider::capacity_boost(*amount));
```

Comment:
I would recommend renaming this to calculate_capacity instead of capacity_generated, as it calculates how much capacity to issue rather than retrieving capacity that has already been generated.